src/libhirte/common/network.c

```diff
@@ -35,25 +34,25 @@ int get_address(const char *domain, char **ip_address) {
         int status = getaddrinfo(domain, NULL, &hints, &res);
         if (status != 0) {
                 hirte_log_errorf("getaddrinfo failed: %s\n", gai_strerror(status));
-                return 1;
+                return status;
         }
 
-        if (res->ai_family == AF_INET) { // IPv4 address
+        if (res->ai_family == AF_INET) {
                 struct sockaddr_in *ipv4 = (struct sockaddr_in *) res->ai_addr;
                 _cleanup_free_ char *reverse = typesafe_inet_ntop4(ipv4);
-                strncpy(*ip_address, reverse, INET_ADDRSTRLEN);
-                if (ip_address == NULL) {
-                        hirte_log_errorf("AF_INET: Failed to convert the IP address. errno: %d\n", errno);
-                        return 1;
+                if (reverse == NULL) {
+                        freeaddrinfo(res);
+                        return -ENOMEM;
                 }
-        } else if (res->ai_family == AF_INET6) { // IPv6 address
+                *ip_address = steal_pointer(&reverse);
+        } else if (res->ai_family == AF_INET6) {
                 struct sockaddr_in6 *ipv6 = (struct sockaddr_in6 *) res->ai_addr;
                 _cleanup_free_ char *reverse = typesafe_inet_ntop6(ipv6);
-                strncpy(*ip_address, reverse, INET6_ADDRSTRLEN);
-                if (ip_address == NULL) {
-                        hirte_log_errorf("AF_INET6: Failed to convert the IP address. errno: %d\n", errno);
-                        return 1;
+                if (reverse == NULL) {
+                        freeaddrinfo(res);
+                        return -ENOMEM;
                 }
+                *ip_address = steal_pointer(&reverse);
         }
```

Comment:
Do we need an else block here if not IP4 and Not IP6?

Reply:
Good question. I don't think we should end up there, but if we do, its helpful to have at least a log or so. Added it - although I am not sure what error code to provide, decided for `EFAULT` for now.